AniJ98/ajagadis-akamai-linode
# Support for Adding Additional Reserved IP to Existing Linodes

📝 Description
This PR introduces functionality and tests for creating Linodes with reserved IP addresses and adding additional reserved IPs to existing Linodes in the Linodego client. These changes are necessary to provide comprehensive support for managing Linodes with reserved IP addresses through the Linode API. The changes include:

1. Implementation of core operations:
 * Creating a Linode with a reserved IP
 * Adding an additional reserved IP to an existing Linode

2. Test coverage:
 * TestInstance_CreateWithReservedIPAddress: Tests creating a Linode with a reserved IP.
 * TestInstance_CreateWithReservedIPAddressVariants: Tests various scenarios for creating a Linode with a reserved IP.
 * TestInstance_AddReservedIPToInstance: Tests adding a reserved IP to an existing Linode.
 * TestInstance_AddReservedIPToInstanceVariants: Covers different cases for adding a reserved IP to an existing Linode.

✔️ How to Test

What are the steps to reproduce the issue or verify the changes?



1. Ensure you have a valid Linode API token. Set up your environment: 

`export LINODE_TOKEN="your_token_here" `

   Additionally you need to set the following if you want to test it in a different environment: 

   ```
 export LINODE_URL="https://api.linode.com"
    export LINODE_API_VERSION="v4beta"
```

2. Navigate to the test directory within the linodego project. Update the LINODE_TOKEN in the Makefile: LINODE_TOKEN="your_token_here"

    Run the tests using one of the following commands: To run all integration tests: 
    `make testint`

    To run a specific test:
    `go test -v ./integration -run TestInstance_CreateWithReservedIPAddressVariants`

Verify the test output for any failures or unexpected behavior.

Note:

Ensure that you have enough quota to reserve IP before adding it to an existing linode. Ensure the IPMAX limit is enough to add additional reserved IP addresses to the linode.
